patches/tModLoader/Terraria/Audio/LegacyAudioSystem.cs.patch

```diff
@@ -47,12 +47,21 @@
  			Engine.Update();
  			AudioTracks = new IAudioTrack[90];
  			TrackNamesByIndex = new Dictionary<int, string>();
-@@ -114,6 +_,8 @@
- 				yield return null;
+@@ -115,6 +_,8 @@
  			}
  		}
-+
-+		internal Cue GetCueInternal(string cueName) => SoundBank.GetCue(cueName);
  
++		internal Cue GetCueInternal(string cueName) => SoundBank.GetCue(cueName);
++
  		public void LoadCue(int cueIndex, string cueName) {
  			CueAudioTrack cueAudioTrack = new CueAudioTrack(SoundBank, cueName);
+ 			TrackNamesByIndex[cueIndex] = cueName;
+@@ -244,7 +_,7 @@
+ 		}
+ 
+ 		public void UpdateCommonTrackTowardStopping(int i, float totalVolume, ref float tempFade, bool isMainTrackAudible) {
+-			if (!WaveBank.IsPrepared)
++			if (!WaveBank.IsPrepared || AudioTracks[i] is null) // should never really be null, save for some in-between loading if music is forced to play before mod loading is complete
```

Comment:
In my testing, it appeared that it could be temporarily null if you load a mod and it brings you to a ModMenu with custom music and attempts to play that music before mod loading is finished. It still plays fine once loading is complete.